### ISSUE TITLE
Setting DocumentWriter options does not make any difference in serialization

### ISSUE THREAD

**Issue by rmeshksar** [graphql-dotnet/graphql-dotnet#2839]
## Description

We are migrating from GraphQL 3.2 to 4.7 and I noticed setting the options of the DocumentWriter does not have any impact on the response serialization. 

### Steps to reproduce

Consider the following code:
```csharp
void Main()
{
	var schema = new Schema { Query = new TestQuery() };

	var exec = new DocumentExecuter();

	var result = exec.ExecuteAsync(options =>
			{
				options.Schema = schema;
				options.Query = "{ test { thisIsAString, thisIsADateTime } }";
			}
	).Result;

	var writer = new DocumentWriter(options =>
	{
		options.Converters.Add(new IsoDateTimeConverter()
		{
		    DateTimeFormat = "yyyy-MMM-dd"
		});

		options.ContractResolver = new ExecutionResultContractResolver(new ErrorInfoProvider())
		{
			NamingStrategy = new KebabCaseNamingStrategy(true, false, false)
		};
	});
	
	writer.WriteToStringAsync(result).Dump();

}

public class TestQuery : ObjectGraphType
{
	public TestQuery()
	{
		Name = "testQuery";
		Description = "Test description";

		Field<TestResponseType>(
			name: "test",
			resolve: context =>
			{
				return new TestResponse();
			}
		);
	}
}

public class TestResponseType : ObjectGraphType<TestResponse>
{
	public TestResponseType()
	{
		Name = "TestReponse";
		Field<StringGraphType>("ThisIsAString");
		Field<DateTimeGraphType>("ThisIsADateTime");
	}
}

public class TestResponse
{
	public string ThisIsAString { get; set; }
	public DateTime ThisIsADateTime { get; set; }
	public TestResponse()
	{
		ThisIsAString = "String Value";
		ThisIsADateTime = new DateTime(2022, 1, 4, 10, 20, 30);
	}
}

```

### Expected result

```javascript
{"data":{"test":{"this-is-a-string":"String Value","this-is-a-date-time":"2022-Jan-04"}}}
```

### Actual result
```javascript
{"data":{"test":{"thisIsAString":"String Value","thisIsADateTime":"2022-01-04T10:20:30"}}}
```

It works as expected on version 3.2.0

**Comment by Shane32** [comment]
Pretty sure this is due to the `ExecutionResultJsonConverter` not respecting the selected default `NamingStrategy` for the property names of child nodes of `data`.  I'm not sure the best way to fix it.  Best I can tell, we have to pass the selected naming strategy from the contract resolver to the constructor of the converter.

**Comment by Shane32** [comment]
There's actually two issues going on at the same time here.  Serialization of scalars within GraphQL.NET v4 now converts non-numeric values to strings "internally" so the JSON serializer is not seeing a date but rather seeing a string.

See https://graphql-dotnet.github.io/docs/migrations/migration4#scalar-deserialization-type-enforcement

You can override the `DateTimeGraphType` with your own implementation to either (a) change serialization of dates within GraphQL.NET, or (b) pass the raw DateTime value to the serializer to let it serialize it as desired.  You will not need to modify all of your code to use the new graph type.

See: https://graphql-dotnet.github.io/docs/getting-started/custom-scalars#replacing-built-in-scalar-types

**Comment by Shane32** [comment]
To fix the date issue, I wrote this, which works:

```cs
public class MyDateTimeGraphType : DateTimeGraphType
{
    public MyDateTimeGraphType()
    {
        Name = "DateTime";
    }

    public override object Serialize(object value) => value switch
    {
        DateTime _ => value,
        DateTimeOffset _ => value,
        null => null,
        _ => ThrowSerializationError(value)
    };
}
```

And then in your sample add this line of code right after creating the schema:

```cs
schema.ReplaceScalar(new MyDateTimeGraphType());
```

**Comment by Shane32** [comment]
I created a fix in PR #2841 and created a test to match your code both for Newtonsoft.Json and System.Text.Json.
